### PR TITLE
fix: Apply pointer visual states of pivot items

### LINKED ISSUE #2171: PivotHeaderItem should set PointerOver visual states
## What would you like to be added:
Hovering mouse over `Pivot` headers should show pointer-over highlight.

```xaml
		<Pivot>
			<PivotItem Header="Tab 1"
					   Content="Some content" />
			<PivotItem Header="Tab 2"
					   Content="Some more and better content" />
		</Pivot>
```
## Why is this needed:
https://github.com/unoplatform/private/issues/103

## For which Platform:

- [ ] iOS
- [ ] Android
- [x] WebAssembly
- [ ] WebAssembly renderers for Xamarin.Forms
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #2171

## PR Type

What kind of change does this PR introduce?

- Bugfix

## What is the current behavior?

Visual states not applied on hover/press

## What is the new behavior?

Visual states are correctly applied.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Pivot/Pivot.cs [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
 
 namespace Windows.UI.Xaml.Controls
 {
 	public partial class Pivot : ItemsControl
 	{
+		private static class PivotHeaderItemSelectionStates
+		{
+			public const string Disabled = "Disabled";
+			public const string Unselected = "Unselected";
+			public const string UnselectedLocked = "UnselectedLocked"; // TODO: This is not currently in use.
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/Pivot/Pivot.cs:19]: Anyone knows what this is for?

**Comment by jeromelaban** [comment]
Thanks! Are you able to validate some of the states colors? Pressed/hover will not be easy to validate, but the rest can be.

File: src/Uno.UI/UI/Xaml/Controls/Pivot/Pivot.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/Pivot/Pivot.cs:19]: No, but we may end up replacing this code with the official pivot code when available.

**Comment by Youssef1313** [comment]
> Thanks! Are you able to validate some of the states colors? Pressed/hover will not be easy to validate, but the rest can be.

If you mean manual testing, yes I recall I tested this. Both Uno and UWP changes appearance on hover properly. For pressing, I didn't notice any difference in UWP, and Uno follows that.

**Comment by jeromelaban** [comment]
Got it. It'd be interesting to add an "IsEnabled" state validation, though, as this was not available before.

**Comment by jeromelaban** [comment]
That's curious indeed. It could be that the style does not do anything with the disabled state. This will do then, let's merge this :)

**Review by jeromelaban — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**